File: frontend/src/types/wiki.ts
/** Wiki module types */
export interface WikiFolder {
  id: string
  organization_id: string
  parent_id?: string | null
  name: string
  sort_order: number
  created_by: string
  created_at: string
  updated_at: string
  children?: WikiFolder[] | null
  page_count?: number | null
}

export interface WikiPage {
  id: string
  organization_id: string
  folder_id?: string | null
  title: string
  content?: string | null
  content_html?: string | null
  format: string
  version: number
  is_published: boolean
  created_by: string
  creator_name?: string | null
  last_edited_by?: string | null
  editor_name?: string | null
  created_at: string
  updated_at: string
}

export interface WikiPageVersion {
  id: string
  page_id: string
  version: number
  title: string
  content?: string | null
  content_html?: string | null
  change_note?: string | null
  edited_by: string
  editor_name?: string | null
  created_at: string
}

File: backend/app/schemas/wiki.py
"""Wiki module schemas."""
from __future__ import annotations

from pydantic import BaseModel, Field


class WikiFolderCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    parent_id: str | None = None
    sort_order: int = 0


class WikiFolderUpdate(BaseModel):
    name: str | None = Field(None, max_length=200)
    parent_id: str | None = None
    sort_order: int | None = None


class WikiFolderResponse(BaseModel):
    id: str
    organization_id: str
    parent_id: str | None = None
    name: str
    sort_order: int
    created_by: str
    created_at: str
    updated_at: str
    children: list[WikiFolderResponse] | None = None
    page_count: int | None = None


class WikiPageCreate(BaseModel):
    folder_id: str | None = None
    title: str = Field(..., min_length=1, max_length=500)
    content: str | None = None
    content_html: str | None = None
    format: str = "markdown"
    change_note: str | None = None


class WikiPageUpdate(BaseModel):
    title: str | None = Field(None, max_length=500)
    content: str | None = None
    content_html: str | None = None
    folder_id: str | None = None
    is_published: bool | None = None
    change_note: str | None = None


class WikiPageResponse(BaseModel):
    id: str
    organization_id: str
    folder_id: str | None = None
    title: str
    content: str | None = None
    content_html: str | None = None
    format: str
    version: int
    is_published: bool
    created_by: str
    creator_name: str | None = None
    last_edited_by: str | None = None
    editor_name: str | None = None
    created_at: str
    updated_at: str


class WikiPageVersionResponse(BaseModel):
    id: str
    page_id: str
    version: int
    title: str
    content: str | None = None
    content_html: str | None = None
    change_note: str | None = None
    edited_by: str
    editor_name: str | None = None
    created_at: str
